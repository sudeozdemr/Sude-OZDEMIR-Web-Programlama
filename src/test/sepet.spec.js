import { mount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex"
import Sepetim from "../components/sepet.vue";
describe("Sepetimdeki ürünlerin listelenmesi", () => {

    const localVue = createLocalVue()
    localVue.use(Vuex)
    const mockFunction = jest.fn();
    const cartGridModel = [{
        product: {
            "id": 5,
            "urunAdi": "Acuvue OASYS ® 1-Day 30 lu Kutu",
            "urunAciklama": "Günlük Kontakt Lens ACUVUE OASYS 1-Day silikon hidrojel günlük kontakt lensler en yüksek...",
            "fiyat": 149.90,
            "image": "https://cdn.lensmarket.com/ext/urun/air-optix-hydraglyde-1-l.jpg"
        },
        cart: {
            id: "1213",
            count: 3,
            pid: 5,
            goz: "Sol",
            dia: "14:00",
            bc: "8.40",
            sph: "-12.00"
        }
    }, {
        product: {
            "id": 6,
            "urunAdi": "Acuvue OASYS ® 1-Day 30 lu Kutu",
            "urunAciklama": "Günlük Kontakt Lens ACUVUE OASYS 1-Day silikon hidrojel günlük kontakt lensler en yüksek...",
            "fiyat": 149.90,
            "image": "https://cdn.lensmarket.com/ext/urun/air-optix-hydraglyde-1-l.jpg"
        },
        cart: {
            id: "1214",
            count: 3,
            pid: 5,
            goz: "Sağ",
            dia: "14:00",
            bc: "8.40",
            sph: "-12.00"
        }
    }];
    const wrapper = mount(Sepetim, {
        stubs: ["router-link"],
        localVue,
        store: new Vuex.Store({
            actions: {
                fetchCartGridModel: () => {
                    return new Promise(resolve => {
                        resolve(cartGridModel)
                    })
                },
            },
        })
    })

    describe("Gözükecekler", () => {
        it("Kargo Ücreti text", () => {
            expect(wrapper.text()).toContain("Kargo Ücreti");
        });
        it("Ürün 1'in adı : Acuvue OASYS ® 1-Day 30 lu Kutu :", () => {
            expect(wrapper.text()).toContain("Acuvue OASYS ® 1-Day 30 lu Kutu");
        });

        it("Ürün 1'nin hangi göze göre olduğu : Sol", () => {
            expect(wrapper.text()).toContain("Sol");
        });

        it("Ürün 2'in adı : Acuvue OASYS ® 1-Day 30 lu Kutu :", () => {
            expect(wrapper.text()).toContain("Acuvue OASYS ® 1-Day 30 lu Kutu");
        });

        it("Ürün 2'nin hangi göze göre olduğu : Sağ", () => {
            expect(wrapper.text()).toContain("Sağ");
        });
        it("Toplam Sepet Tutarı : 3*149.90", () => {
            const expectedTotal = Number.parseFloat(3 * 149.90).toFixed(2)
            expect(wrapper.text()).toContain(expectedTotal);
        });
    });

    test("Snapshot Testi", () => {
        expect(wrapper.html()).toMatchSnapshot();
    });
})