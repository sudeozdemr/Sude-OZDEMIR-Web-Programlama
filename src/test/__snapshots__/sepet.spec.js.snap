// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sepetimdeki ürünlerin listelenmesi Snapshot Testi 1`] = `
"<div class=\\"container clearfix\\">
  <div class=\\"siparisToplamBox\\">
    <div class=\\"siparisToplamInfo\\">
      <div style=\\"font-size: 12px;\\">Kargo Ücreti</div>
      <div style=\\"text-align: right;\\"><del style=\\"font-size: 12px;\\">11,90 TL</del><br>0,00 TL
      </div>
      <div style=\\"font-size: 14px; margin-top: 30px;\\"><b>Genel Toplam</b></div>
      <div style=\\"text-align: right; color: orange; margin-top: 30px;\\">
        899.40 TL (KDV Dahil)
      </div>
    </div> <button class=\\"btnSatinAl\\"><img src=\\"src/assets/satinalikon.png\\" alt=\\"yok\\" style=\\"width: 45px; margin-top: -10px; display: inline-block; float: left;\\">SATIN AL
    </button>
  </div>
  <div class=\\"urunSepettekiler clearfix\\">
    <div class=\\"urunBox clearfix\\">
      <div class=\\"silmeButonDiv\\"><button>X</button></div>
      <router-link-stub to=\\"[object Object]\\">
        <div class=\\"urunResimDiv\\"><img src=\\"https://cdn.lensmarket.com/ext/urun/air-optix-hydraglyde-1-l.jpg\\" alt=\\"\\"></div>
      </router-link-stub>
      <div class=\\"urunAciklamaBody clearfix\\">
        <router-link-stub to=\\"[object Object]\\">
          <div class=\\"urunInfo\\">
            <h3 class=\\"urunInfoTitle\\">Acuvue OASYS ® 1-Day 30 lu Kutu</h3>
            <table>
              <thead>
                <tr>
                  <td>Göz</td>
                  <td>Dia</td>
                  <td>Bc</td>
                  <td>Sph (Pwr)</td>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>Sol</td>
                  <td>14:00</td>
                  <td>8.40</td>
                  <td>-12.00</td>
                </tr>
              </tbody>
            </table>
          </div>
        </router-link-stub>
        <div class=\\"urunAdet\\"><select class=\\"urunAdetSelect\\">
            <option value=\\"1\\" selected=\\"selected\\">1 Kutu</option>
            <option value=\\"2\\">2 Kutu</option>
            <option value=\\"3\\">3 Kutu</option>
            <option value=\\"4\\">4 Kutu</option>
          </select></div>
        <div class=\\"urunFiyat\\"><span>449.70</span>
          TL
        </div>
      </div>
    </div>
    <div class=\\"urunBox clearfix\\">
      <div class=\\"silmeButonDiv\\"><button>X</button></div>
      <router-link-stub to=\\"[object Object]\\">
        <div class=\\"urunResimDiv\\"><img src=\\"https://cdn.lensmarket.com/ext/urun/air-optix-hydraglyde-1-l.jpg\\" alt=\\"\\"></div>
      </router-link-stub>
      <div class=\\"urunAciklamaBody clearfix\\">
        <router-link-stub to=\\"[object Object]\\">
          <div class=\\"urunInfo\\">
            <h3 class=\\"urunInfoTitle\\">Acuvue OASYS ® 1-Day 30 lu Kutu</h3>
            <table>
              <thead>
                <tr>
                  <td>Göz</td>
                  <td>Dia</td>
                  <td>Bc</td>
                  <td>Sph (Pwr)</td>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>Sağ</td>
                  <td>14:00</td>
                  <td>8.40</td>
                  <td>-12.00</td>
                </tr>
              </tbody>
            </table>
          </div>
        </router-link-stub>
        <div class=\\"urunAdet\\"><select class=\\"urunAdetSelect\\">
            <option value=\\"1\\" selected=\\"selected\\">1 Kutu</option>
            <option value=\\"2\\">2 Kutu</option>
            <option value=\\"3\\">3 Kutu</option>
            <option value=\\"4\\">4 Kutu</option>
          </select></div>
        <div class=\\"urunFiyat\\"><span>449.70</span>
          TL
        </div>
      </div>
    </div>
  </div>
</div>"
`;
